x-pack/plugins/ml/public/application/jobs/new_job/job_from_map/quick_create_job.ts

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { MapEmbeddable } from '@kbn/maps-plugin/public';
+import type { IUiSettingsClient } from '@kbn/core/public';
+import type { TimefilterContract } from '@kbn/data-plugin/public';
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+import type { Filter, Query } from '@kbn/es-query';
+import type { DataView } from '@kbn/data-views-plugin/public';
+
+import type { MlApiServices } from '../../../services/ml_api_service';
+import { getDataViews } from '../../../util/dependency_cache';
+import {
+  CREATED_BY_LABEL,
+  JOB_TYPE,
+  DEFAULT_BUCKET_SPAN,
+} from '../../../../../common/constants/new_job';
+import { createEmptyJob, createEmptyDatafeed } from '../common/job_creator/util/default_configs';
+import type { JobCreatorType } from '../common/job_creator';
+import { stashJobForCloning } from '../common/job_creator/util/general';
+import { getJobsItemsFromEmbeddable } from './utils';
+import { QuickJobCreatorBase, CreateState } from '../job_from_dashboard';
+import { getDefaultQuery } from '../utils/new_job_utils';
+
+interface VisDescriptor {
+  dashboard: { query: Query; filters: Filter[] };
+  embeddable: { query: Query; filters: Filter[] };
+  dataViewId?: string;
+  sourceDataView?: DataView;
+  geoField: string;
+  splitField: string | null;
+  bucketSpan?: string;
+}
+
+export class QuickJobCreator extends QuickJobCreatorBase {
+  constructor(
+    public readonly kibanaConfig: IUiSettingsClient,
+    public readonly timeFilter: TimefilterContract,
+    public readonly share: SharePluginStart,
+    public readonly mlApiServices: MlApiServices
+  ) {
+    super(kibanaConfig, timeFilter, share, mlApiServices);
+  }
+
+  public async createAndSaveGeoJob({
```

Comment:
I wonder whether the functions should be named the same for the lens and geo versions of these classes? even though they take different parameters.

Ideally these functions would be abstract in the base class, but the params are so different it might not be worth it.

Also, is there a reason why all of these functions take a single object as a parameter rather than a list of parameters?

Reply:
any thoughts on this?